github/commands/pull_request.py

```diff
@@ -74,59 +76,113 @@ def on_select_action(self, idx):
             show_single_line_input_panel(
                 "Enter branch name for PR {}:".format(self.pr["number"]),
-                "{}/{}".format(self.pr["user"]["login"], self.pr["head"]["ref"]),
+                branch_name,
                 self.fetch_and_checkout_pr
             )
         elif idx == 2:
             show_single_line_input_panel(
                 "Enter branch name for PR {}:".format(self.pr["number"]),
-                "{}/{}".format(self.pr["user"]["login"], self.pr["head"]["ref"]),
-                self.create_branch_for_pr
+                branch_name,
+                lambda x: self.create_branch_for_pr(x, ask_set_upstream=True)
             )
         elif idx == 3:
             self.view_diff_for_pr()
         elif idx == 4:
             self.open_pr_in_browser()
 
     def fetch_and_checkout_pr(self, branch_name=None):
-        self.window.status_message("Fetching PR commit...")
-        self.git(
-            "fetch",
-            self.pr["head"]["repo"]["clone_url"],
-            self.pr["head"]["ref"]
-        )
-
         if branch_name:
-            self.window.status_message("Creating local branch for PR...")
-            self.git(
-                "branch",
-                branch_name,
-                self.pr["head"]["sha"]
-            )
+            self.create_branch_for_pr(branch_name, checkout=True, ask_set_upstream=True)
+        else:  # detached HEAD
+            self.checkout_detached()
 
+    def checkout_detached(self):
         self.window.status_message("Checking out PR...")
-        self.checkout_ref(branch_name or self.pr["head"]["sha"])
+
+        clone_url = self.pr["head"]["repo"]["clone_url"]
+        ssh_url = self.pr["head"]["repo"]["ssh_url"]
+        ref = self.pr["head"]["ref"]
+
+        if self.base_remote_url.startswith("https://"):
+            url = clone_url
+        elif self.base_remote_url.startswith("git@"):
+            url = ssh_url
+        else:
+            raise RuntimeError("Cannot recognize url {}.".format(self.base_remote_url))
+
+        self.git("fetch", url, ref)
+        self.checkout_ref(ref)
         util.view.refresh_gitsavvy_interfaces(self.window, refresh_sidebar=True)
 
-    def create_branch_for_pr(self, branch_name):
-        self.window.status_message("Fetching PR commit...")
-        self.git(
-            "fetch",
-            self.pr["head"]["repo"]["clone_url"],
-            self.pr["head"]["ref"]
-        )
+    def create_branch_for_pr(self, branch_name, checkout=False, ask_set_upstream=False):
+        if not branch_name:
+            return
 
         self.window.status_message("Creating local branch for PR...")
-        self.git(
-            "branch",
-            branch_name,
-            self.pr["head"]["sha"]
-        )
+        set_upstream = False
+
+        remotes = self.get_remotes()
+        owner = self.pr["head"]["repo"]["owner"]["login"]
+        if owner == self.base_remote.owner:
+            owner = self.base_remote_name
```

Comment:
```suggestion
            owner = self.get_integrated_remote_name()
```

Can you explain what this does. First, owner is `self.pr["head"]["repo"]["owner"]["login"]` which is probably a GitHub username? Here we change the `owner` to `base_remote_name`, which is the local remote name(?), so we have for example `owner == origin` here?

Reply:
`self.pr["head"]["repo"]["owner"]["login"]` could be an github user/organization name. For example, it could be `timbrel` if a PR is coming from a branch of `GitSavvy`. It is checking if the `base_remote` owner is the same as the PR's owner. 

In most cases, `self.base_remote.owner` would be the username of the PR submitter and `self.base_remote_name` would be `"origin".`